server.js

```diff
@@ -1,269 +1,360 @@
+    {
+      method: 'POST',
+      path: '/v0/subscribe',
+      config: {
+        auth: {
+          strategy: 'fxa-oauth',
+          scope: ['notify']
+        },
+        validate: {
+          payload: {
+            notify_url: joi.string().required(),
+            filter: joi.object().keys({
+              uid: joi.string().optional(),
+              rid: joi.string().optional(),
+              iss: joi.string().optional(),
+              typ: joi.string().optional()
+            }).optional(),
+            ttl: joi.number().min(0).optional(),
+            pos: joi.string().optional()
+          }
+        }
+      },
+      handler: function (req, reply) {
+        var sub = subscriptions.add(req.payload)
+        reply({
+          id: sub.id
+        })
       }
-      reply(sub)
-    }
-  },
-  {
-    method: 'DELETE',
-    path: '/v0/subscription/{id}',
-    config: {
-      auth: {
-        strategy: 'fxa-oauth',
-        scope: ['notify']
+    },
+    {
+      method: 'GET',
+      path: '/v0/subscription/{id}',
+      config: {
+        auth: {
+          strategy: 'fxa-oauth',
+          scope: ['notify']
+        }
+      },
+      handler: function (req, reply) {
+        var sub = subscriptions.get(req.params.id)
+        reply(sub || boom.notFound(req.params.id))
       }
     },
-    handler: function (req, reply) {
-      subscriptions.remove(req.params.id)
-      reply({})
-    }
-  },
-  {
-    method: 'GET',
-    path: '/v0/subscription/{id}/events',
-    config: {
-      auth: {
-        strategy: 'fxa-oauth',
-        scope: ['notify']
+    {
+      method: 'POST',
+      path: '/v0/subscription/{id}',
+      config: {
+        auth: {
+          strategy: 'fxa-oauth',
+          scope: ['notify']
+        },
+        validate: {
+          payload: {
+            pos: joi.string().optional(),
+            notify_url: joi.string().optional()
+          }
+        }
       },
-      validate: {
-        query: {
-          pos: joi.string().optional(),
-          num: joi.number().min(1).max(1000).optional()
+      handler: function (req, reply) {
+        var sub = subscriptions.get(req.params.id)
+        if (sub) {
+          if (req.payload.pos) { sub.pos = req.payload.pos }
+          if (req.payload.notify_url) { sub.notify_url = req.payload.notify_url }
         }
+        reply(sub || boom.notFound(req.params.id))
       }
     },
-    handler: function (req, reply) {
-      var sub = subscriptions.get(req.params.id)
-      var pos = req.query.pos || '0'
-      var num = req.query.num || 1000
-      if (sub) {
-        reply(db.read(pos, num, sub.filter))
-      }
-      else {
+    {
+      method: 'DELETE',
+      path: '/v0/subscription/{id}',
+      config: {
+        auth: {
+          strategy: 'fxa-oauth',
+          scope: ['notify']
+        }
+      },
+      handler: function (req, reply) {
+        subscriptions.remove(req.params.id)
         reply({})
```

Comment:
I had an internal debate about that in my head, but the timeout expired so I skipped it :)